src/main/java/TicTacToe.java

```diff
@@ -0,0 +1,327 @@
+            }
+        }
+
+        System.out.println("Enter player name");
+        player1 = new Player(getUserInput(),2,0,new HumanMove(this));
+
+        player2 = new Player("",1,1,new SimpleMoveStrategy(this));
+        System.out.println("\nHuman player " + player1.getName() + " vs Computer Player " + player2.getName() + ":" ) ;
+    }
+
+    public boolean setMove(int move, int p_type)
+    {
+        int x_coord = (move-1)/3;
+        int y_coord = (move-1)%3;
+
+        if( board[x_coord][y_coord] == 0 )
+            {
+                board[x_coord][y_coord] = p_type ;
+                return true ;
+            }
+        else
+            {
+                System.out.println("Invalid move");
+                return false ;
+            }
+    }
+
+    private enum WinConfig {
+        DRAW, WIN, NONE
+    }
+
+    private WinConfig isWinningConfig()
+    {
+        WinConfig w = WinConfig.WIN ;
+        // rows
+        for( int i = 0 ; i < N ; i ++ )
+        {
+            if( (board[i][0] != 0) && (board[i][0] == board[i][1]) && (board[i][0] == board[i][2] ) )
+            {
+                return w ;
+            }
+        }
+        // columns
+        for( int i = 0 ; i < N ; i ++ )
+        {
+            if( (board[0][i] != 0) && (board[0][i] == board[1][i]) && (board[0][i] == board[2][i] ) )
+            {
+                return w ;
+            }
+        }
+        // diags
+        if( (board[0][0] != 0) && (board[0][0] == board[1][1]) && (board[0][0] == board[2][2] ) )
+        {
+            return w ;
+        }
+
+        if( (board[2][0] != 0) && (board[2][0] == board[1][1]) && (board[2][0] == board[0][2] ) )
+        {
+            return w ;
+        }
+
+        // draw
+        w = WinConfig.DRAW ;
+        for( int i = 0 ; i < N ; i ++ )
+            for( int j = 0 ; j < N ; j ++ )
+                {
+                    if( board[i][j] == 0 )
+                        w = WinConfig.NONE ;
+                }
+        return w ;
+
+    }
+
+    private String getRowString(int row)
+    {
+        String s = "" ;
+        for( int i = 0 ; i < N ; i ++ )
+        {
+            switch(board[row][i]) {
+            case 0: s += " " ;
+                break ;
+            case 1: s += "O" ;
+                break ;
+            case 2: s += "X" ;
+            }
+
+            if( i != N-1 )
+                {
+                    s += " | " ;
+                }
+        }
+
+            s += String.format("%" + HSPACE + "s", "");
+
+         for( int i = 0 ; i < N ; i ++ )
+          {
+            s += row*3+i+1 ;
+
+            if( i == N-1 ) {
+                s += "\n";
+            }
+            else {
+                s += " | " ;
+            }
+          }
+        return s;
+    }
+
+    public String toString()
+    {
+        String s = "";
+        // iterate through the rows
+        for( int i = 0 ; i < N ; i ++ ) {
+            s += getRowString(i);
+        }
+        return s;

```

Comment:
Should really use a StringBuilder.
If we had a Board object that represented the board we could hide this complexity to get the Board string.